### PR TITLE
Nutch 1.19

### PR DESCRIPTION
**Pull request by zhuomingao**
Nutch 1.19 upgrade

### CODE REVIEW

File: data_nutch_data_nci/conf/nutch-site.xml [hunk 1]
```diff
@@ -45,7 +45,7 @@
 
 	<property>
 		<name>db.fetch.retry.max</name>
-		<value>30</value>
+		<value>10</value>
 		<description>NCI OVERRIDE: Things on the network can be sketchy, so I am upping this to a high value so that we do not accidently remove things that are having issues.  A value of 30 *should* basically mean that we should retry the item for about a month before skipping it.</description>
```
Comment (blairlearn) [inline data_nutch_data_nci/conf/nutch-site.xml:49]: Is this a deliberate change?  The comment says it should be a month.

File: data_nutch_data_nci/conf/nutch-site.xml [hunk 2]
```diff
@@ -63,19 +63,23 @@
 
 	<property>
 		<name>http.timeout</name>
-		<value>10000</value>
+		<value>20000</value>
 		<description>NCI OVERRIDE: Upped to 20sec from 10sec to allow time for www.cancer.gov to generate its sitemap.xml file. 20151113 - it seems 20sec is not long enough, so reverting to 10sec</description>
```
Comment (blairlearn) [inline data_nutch_data_nci/conf/nutch-site.xml:67]: Here the comment says it's reverted to 10 seconds, but the change moves it to 20.

File: data_nutch_data_nci/conf/nutch-site.xml [hunk 3]
```diff
@@ -63,19 +63,23 @@
...
 	
+
+
+
+
 	<property>
 	  <name>fetcher.threads.fetch</name>
-	  <value>100</value>
+	  <value>50</value>
 	  <description>NCI OVERRIDE: Upping this to 100</description>
```
Comment (blairlearn) [inline data_nutch_data_nci/conf/nutch-site.xml:77]: More disagreement between the configuration and the adjoining comments.

File: data_nutch_data_nci/conf/nutch-site.xml [hunk 4]
```diff
@@ -84,56 +88,61 @@
...
 
 	<!-- If fetcher.threads.per.queue == 1 then comment the lines below -->
-	<property>
+	<!--<property>
 	  <name>fetcher.server.min.delay</name>
 	  <value>0.3</value>
 	  <description>NCI OVERRIDE: We need to ignore crawl delays in robots.txt for many Drupal sites which are set at 10 sec.  Setting too many threads will kill sites.  This setting acts like fetcher.server.delay, but </description>
 	</property>
 
+	-->
```
Comment (blairlearn) [inline data_nutch_data_nci/conf/nutch-site.xml:97]: Should this instead be deleted?  Or was this a temporary change that didn't get reverted?

File: data_nutch_data_nci/conf/regex-normalize.xml
Comment (blairlearn) [inline data_nutch_data_nci/conf/regex-normalize.xml:1]: Was this file simply reformatted by mistake?  All of the changes appear to be indents.

File: data_nutch_data_nci/conf/seeds.txt [hunk 5]
```diff
@@ -1,24 +1,25 @@
 ## Main CGov Site to Start
-https://www.cancer.gov/sitemap.xml	nutch.fetchInterval=86400
```
Comment (blairlearn) [inline data_nutch_data_nci/conf/seeds.txt:2]: Wouldn't you want to keep `sitemap.xml`?  The microsites (e.g. `/nano`) don't otherwise appear in this list.

**Review by blairlearn — CHANGES_REQUESTED** [review]
Probably most of this should remain as-is, but I don't know, so I have to ask.

File: data_nutch_data_nci/conf/.DS_Store
[the remark this answers is not in the document]
Comment (zhuomingao) [reply data_nutch_data_nci/conf/.DS_Store:1]: add .gitignore  for **/.DS_Store

File: data_nutch_data_nci/conf/crawl-urlfilter.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+# Each non-comment, non-blank line contains a regular expression
+# prefixed by '+' or '-'.  The first matching pattern in the file
+# determines whether a URL is included or ignored.  If no pattern
+# matches, the URL is ignored.
+
+# skip file:, ftp:, & mailto: urls
+-^(file|ftp|mailto|https):
+
+# skip image and other suffixes we can't yet parse
+-\.(gif|GIF|jpg|JPG|ico|ICO|css|sit|eps|wmf|rtf|zip|ppt|mpg|xls|gz|rpm|tgz|mov|MOV|exe|mp3|rss|xml|doc|pdf|txt|DOC|PDF|TXT)$
```
[the remark this answers is not in the document]
Comment (zhuomingao) [reply data_nutch_data_nci/conf/crawl-urlfilter.txt]: removed this file

File: data_nutch_data_nci/conf/nutch-site.xml (hunk 1 above)
Comment (zhuomingao) [reply data_nutch_data_nci/conf/nutch-site.xml]: update to be 30

File: data_nutch_data_nci/conf/nutch-site.xml (hunk 2 above)
Comment (zhuomingao) [reply data_nutch_data_nci/conf/nutch-site.xml]: updated the comment to be 20000

File: data_nutch_data_nci/conf/nutch-site.xml (hunk 3 above)
Comment (zhuomingao) [reply data_nutch_data_nci/conf/nutch-site.xml]: updated the comment to use 50

File: data_nutch_data_nci/conf/nutch-site.xml (hunk 4 above)
Comment (zhuomingao) [reply data_nutch_data_nci/conf/nutch-site.xml:97]: Keeping this comment is deliberate.

File: data_nutch_data_nci/conf/seeds.txt (hunk 5 above)
Comment (zhuomingao) [reply data_nutch_data_nci/conf/seeds.txt:2]: https://www.cancer.gov/sitemap.xml contains below, SWS doesn't need dictionaries etc.
<sitemap>
<loc>https://www.cancer.gov/sitemaps/pageinstructions.xml</loc>
<lastmod>2019-06-18T07:59:48.4052333-04:00</lastmod>
</sitemap>
<sitemap>
<loc>https://www.cancer.gov/sitemaps/dictionaries.xml</loc>
<lastmod>2019-06-18T07:59:48.4052333-04:00</lastmod>
</sitemap>
<sitemap>
<loc>https://www.cancer.gov/sitemaps/dynamiclistingpages.xml</loc>
<lastmod>2019-06-18T07:59:48.4052333-04:00</lastmod>
</sitemap>
<sitemap>
<loc>https://www.cancer.gov/sitemaps/clinicaltrials.xml</loc>
<lastmod>2019-06-18T07:59:48.4052333-04:00</lastmod>
</sitemap>

File: data_nutch_data_nci/conf/crawl-urlfilter.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (blairlearn) [reply data_nutch_data_nci/conf/crawl-urlfilter.txt]: Why did you remove it? If this part of the configuration, then it belongs.

[the remark this answers is not in the document]
Comment (zhuomingao) [reply data_nutch_data_nci/conf/crawl-urlfilter.txt]: I think this was used in nutch 1.17, but in current nutch 1.19, there is no plugin using this config file.

File: data_nutch_data_nci/conf/seeds.txt (hunk 5 above)
Comment (blairlearn) [reply data_nutch_data_nci/conf/seeds.txt:2]: That's what I'm getting at.  This doesn't have anything to point nutch to www.cancer.gov/nano, www.cancer.gov/pediatric-adult-rare-tumor, and the other microsites which are listed in the main sitemap.xml.

(On the other hand, the file **does** contain a reference to https://cancergenome.nih.gov/, and that site has been a redirect to https://www.cancer.gov/ccg/research/genome-sequencing/tcga for several years.)

**Review by blairlearn — CHANGES_REQUESTED** [review]
See the remaining unresolved review comments.  There are several items I'm still concerned about.

**Review by blairlearn — APPROVED** [review]

**Pull request merged**